src/functions/containers/FunctionEditPage.tsx

```diff
@@ -1,21 +1,107 @@
 // Libraries
-import React, {FC} from 'react'
+import React, {FC, lazy, Suspense, useContext, useEffect, useState} from 'react'
+import {useHistory, useParams} from 'react-router-dom'
+import {useSelector} from 'react-redux'
 
 // Components
-import {Page} from '@influxdata/clockface'
+import {
+  Page,
+  RemoteDataState,
+  SpinnerContainer,
+  TechnoSpinner,
+} from '@influxdata/clockface'
+import FunctionHeader from 'src/functions/components/FunctionHeader'
+import FunctionForm from 'src/functions/components/FunctionForm'
+import FunctionListProvider, {
+  FunctionListContext,
+} from 'src/functions/context/function.list'
+
+const FluxMonacoEditor = lazy(() =>
+  import('src/shared/components/FluxMonacoEditor')
+)
 
 // Utils
 import {pageTitleSuffixer} from 'src/shared/utils/pageTitles'
+import {getOrg} from 'src/organizations/selectors'
 
 const FunctionEditPage: FC = () => {
+  const history = useHistory()
+  const {id: functionID} = useParams<{id: string}>()
+  const {id: orgID} = useSelector(getOrg)
+
+  const [name, setName] = useState('New Function')
+  const [description, setDescription] = useState('')
+  const [params, setParams] = useState('')
+  const [script, setScript] = useState('')
+
+  const {functionsList} = useContext(FunctionListContext)
+  useEffect(() => {
+    const {name, script, description} = functionsList[functionID]
+    // TODO what if functionID no exist??
+    setName(name)
+    setScript(script)
+    setDescription(description)
+  }, [functionsList, functionID, setName, setScript, setDescription])
```

Comment:
i'm circling here, looking for words. but this isn't what you want right? if you have the page open in two tabs, or a server push updates the list of functions because someone is working on the same page state, you dont want to _reload_ the function into the form's state right? but use this more like a `componentDidMount`?